src/cdk/tree/tree.ts

```diff
@@ -94,6 +102,74 @@ type RenderingData<T> =
       renderNodes: readonly T[];
     };
 
+interface RenderNode<T, K> {
+  context: CdkTreeNodeOutletContext<T>;
+  template: TemplateRef<CdkTreeNodeOutletContext<T>>;
+  key: K;
+}
+
+class NodeOutletTemplateContext<T, K> {
+  $implicit: Signal<RenderNode<T, K>[]>;
+
+  constructor(nodes: Signal<RenderNode<T, K>[]>) {
+    this.$implicit = nodes;
+  }
+}
+
+/** @docs-private */
+@Directive({
```

Comment:
If this is private, do we need a separate component for it?

Reply:
I'm not really sure of another way to hook into the onInit for the directive here, though I think ultimately removing `mostRecentTreeNode`  would solve this issue (e.g. doing data passing through injection)